pilot/pkg/config/file/store.go

```diff
@@ -78,7 +81,20 @@ func (s *KubeSource) Get(typ config.GroupVersionKind, name, namespace string) *c
 }
 
 func (s *KubeSource) List(typ config.GroupVersionKind, namespace string) ([]config.Config, error) {
-	return s.inner.List(typ, namespace)
+	configs, err := s.inner.List(typ, namespace)
+	if err != nil {
+		return nil, err
+	}
+	if namespace == model.NamespaceAll && s.namespacesFilter != nil {
+		var out []config.Config
```

Comment:
List if pretty big sometimes.. can/should we in-place filter the list instead of copy?